custom_components/localtuya/__init__.py

```diff
@@ -206,6 +210,15 @@ def _shutdown(event):
     except Exception:  # pylint: disable=broad-except
         _LOGGER.exception("failed to set up discovery")
 
+    async def _async_reconnect(now):
+        """Try connecting to devices not already connected to."""
+        for entry in hass.data[DOMAIN]:
+            device = entry[TUYA_DEVICE]
```

Comment:
`entry` is a string (the key of hass.data[DOMAIN] which is entry_id). Probably the reason for the exception `TypeError: string indices must be integers
`

Reply:
Yeah, that was a mistake on my part. Should be fixed now.